homeassistant/components/nx584/alarm_control_panel.py

```diff
@@ -46,27 +51,47 @@ def setup_platform(hass, config, add_entities, discovery_info=None):
     url = f"http://{host}:{port}"
 
     try:
-        add_entities([NX584Alarm(hass, url, name)])
+        alarmclient = client.Client(url)
+        await hass.async_add_executor_job(alarmclient.list_zones)
     except requests.exceptions.ConnectionError as ex:
-        _LOGGER.error("Unable to connect to NX584: %s", str(ex))
-        return
+        _LOGGER.error(
+            "Unable to connect to %(host)s: %(reason)s", dict(host=url, reason=ex),
+        )
+        raise PlatformNotReady
+
+    entity = NX584Alarm(name, alarmclient, url)
+    async_add_entities([entity])
+
+    platform = entity_platform.current_platform.get()
+
+    platform.async_register_entity_service(
+        SERVICE_BYPASS_ZONE,
+        {
+            vol.Required(ATTR_ENTITY_ID): cv.entity_ids,
```

Comment:
Remove the entity_id item. The service helper will generate a schema containing the entity_id item for us.

Reply:
Okay.  Removed from both async_register_entity_service calls.

Reply:
Had to add a directive for the flake8 linter.  Assume that is ok?  Black would reformat in Code then flake8 would complain when trying to commit